client/src/services/ServiceWorker/Controllers/SigningController.js

```diff
@@ -0,0 +1,171 @@
+import EventEmitter from 'events';
+import {
+    DeployUtil,
+    encodeBase16,
+    formatMessageWithHeaders,
+} from 'casper-js-sdk';
+import _pick from 'lodash-es/pick';
+import { getDeployArgs, getDeployPayment, getDeployType } from '../utils';
+
+const SIGNING_STATUS = {
+    unsigned : 'Unsigned',
+    signed : 'Signed',
+    failed : 'Failed'
+}
+
+
+class SigningController extends EventEmitter{
+    unsignedDeploy = {};
+    unsignedMessage = {};
+    popupController;
+    accountController;
+
+    constructor(popupController, accountController) {
+        super()
+        this.popupController = popupController;
+        this.accountController = accountController;
+    }
+
+    getActivePublicKey = async ({ origin }) => {
+        const isConnected = await this.popupController.isConnected({ origin });
+        if (!isConnected) {
+            throw new Error('This site is not connected');
+        } 
+
+        return this.accountController.getCurrentPublicKey();
+    }
+
+    signDeploy = async (
+        deployJson,
+        signingPublicKeyHex,
+        targetPublicKeyHex,
+        origin,
+    ) => {
+        // const isConnected = await this.popupController.isConnected({ origin });
+        // if (!isConnected) {
+        //     throw new Error('This site is not connected');
+        // }
+
+        let innerDeploy = DeployUtil.deployFromJson(deployJson.deploy);
+        if (!innerDeploy.ok) {
+            innerDeploy.mapErr(err => {
+                throw new Error(err.message);
+            });
+        }
+
+        this.unsignedDeploy = {
+            status: SIGNING_STATUS.unsigned,
+            deploy: innerDeploy.unwrap(),
```

Comment:
Can you please comment why we need `unwrap` here? Thanks.
Ref: 
https://casper-ecosystem.github.io/casper-js-sdk/latest/modules/_lib_deployutil_.html#deployfromjson

Reply:
We will get the deploy hash when using unwrap function